pkg/util/docker/network_test.go

```diff
@@ -57,48 +57,126 @@ func TestFindDockerNetworks(t *testing.T) {
-						Gateway: "172.18.0.1",
+					"bridge": &dockernetwork.EndpointSettings{
+						Gateway: "172.17.0.1",
 					},
-					"eth0": &dockernetwork.EndpointSettings{
-						Gateway: "172.0.0.4/24",
+					"test": &dockernetwork.EndpointSettings{
+						Gateway: "172.18.0.1",
 					},
 				},
 			},
 			routes: detab(`
-                Iface   Destination Gateway     Flags   RefCnt  Use Metric  Mask        MTU Window  IRTT
-                eth0    00000000    010012AC    0003    0   0   0   00000000    0   0   0
-
-                eth0    000012AC    00000000    0001    0   0   0   0000FFFF    0   0   0
+				Iface	Destination	Gateway 	Flags	RefCnt	Use	Metric	Mask		MTU	Window	IRTT
+				eth0	00000000	010011AC	0003	0	0	0	00000000	0	0	0
+				eth0	000011AC	00000000	0001	0	0	0	0000FFFF	0	0	0
+				eth1	000012AC	00000000	0001	0	0	0	0000FFFF	0	0	0
             `),
 			dev: detab(`
-                Inter-|   Receive                                                |  Transmit
-                 face |bytes    packets errs drop fifo frame compressed multicast|bytes    packets errs drop fifo colls carrier compressed
-                  eth0:    1111       2    0    0    0     0          0         0     1024      80    0    0    0     0       0          0
-                    lo:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
-            `),
+				Inter-|   Receive                                                |  Transmit
+				 face |bytes    packets errs drop fifo frame compressed multicast|bytes    packets errs drop fifo colls carrier compressed
+				    lo:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
+				  eth0:     648       8    0    0    0     0          0         0        0       0    0    0    0     0       0          0
+				  eth1:    1478      19    0    0    0     0          0         0      182       3    0    0    0     0       0          0`),
 			networks: []dockerNetwork{
-				dockerNetwork{iface: "eth0", dockerName: "eth0"},
-				dockerNetwork{iface: "eth0", dockerName: "isolated_nw"},
+				dockerNetwork{iface: "eth0", dockerName: "bridge"},
+				dockerNetwork{iface: "eth1", dockerName: "test"},
 			},
-			stat: &NetworkStat{
-				BytesRcvd:   1111,
-				PacketsRcvd: 2,
-				BytesSent:   1024,
-				PacketsSent: 80,
+			stat: &ContainerNetStats{
+				Stats: []*InterfaceNetStats{
+					&InterfaceNetStats{
+						NetworkName: "bridge",
+						BytesRcvd:   648,
+						PacketsRcvd: 8,
+						BytesSent:   0,
+						PacketsSent: 0,
+					},
+					&InterfaceNetStats{
+						NetworkName: "test",
+						BytesRcvd:   1478,
+						PacketsRcvd: 19,
+						BytesSent:   182,
+						PacketsSent: 3,
+					},
+				},
+			},
+			summedStat: &InterfaceNetStats{
+				BytesRcvd:   2126,
+				PacketsRcvd: 27,
+				BytesSent:   182,
+				PacketsSent: 3,
 			},
 		},
+		/* TODO: where is this example from?
```

Comment:
@conorbranagan This test case looks strange to me:
  - the IP in the inspect does not match the hex in /proc/net/dev
  - as far as I know, if several networks are attached to one container, it should have several interfaces, should't `isolated_nw` be `eth1`? Where did you see that case?

Reply:
Honestly can't remember where this case came from, feel free to clean up if you have better ones covering everything.